### DISCUSSION TITLE
custom middleware for server routes in nuxt3

### DISCUSSION THREAD

**Discussion by Yukie-Billal** [nuxt/nuxt#25522] [Questions] [answered]
guys, i want make middleware for my server routes, give me an example pleasee. 
is nuxt `/middleware` can work for server routes to??

## here my server routes code `/server/[...].ts`

```js
import {useBase, createRouter, defineEventHandler} from 'h3'
const router = createRouter()

router.get('/users', defineEventHandler(users.get))

export default useBase('/api', router.handler)
```

for more info: i do migration from express-js to fullstack nuxt

**Comment by AloisSeckar** [comment, marked as answer]
Here is a simple implementation:
https://github.com/AloisSeckar/demos-nuxt/blob/main/nuxt-middleware/server/middleware/server.ts

Server middleware files belong to `/server/middleware` directory. You just implement and export `defineEventHandler` method and thats it. Just be aware they run before **EACH** server request, afaik it is not possible to use them route-based.

**Reply by Yukie-Billal** [reply]
is this good to set manually if statement for route middleware? 

```js
let canAccess = true
if (event.node.req.url === '/custom-access') canAccess = jwtVerify('super', event.node.req.headers['authorization'])

if (!canAccess) throw createError({
  statusCode: 403,
  statusMessage: 'Forbidden'
})
```

is there option to target some routes ?
